scapy/data.py

```diff
@@ -290,17 +292,69 @@
 }
 
 
+def scapy_data_cache(name):
+    # type: (str) -> Callable[[DecoratorCallable], DecoratorCallable]
+    """
+    This decorator caches the loading of 'data' dictionaries, in order to reduce
+    loading times.
+    """
+    from scapy.main import SCAPY_CONFIG_FOLDER
+    if SCAPY_CONFIG_FOLDER is None:
+        # Cannot cache.
+        return lambda x: x
+    cachepath = SCAPY_CONFIG_FOLDER / "cache" / name
+
+    def _cached_loader(func, name=name):
+        # type: (DecoratorCallable, str) -> DecoratorCallable
+        def load(filename=None):
+            # type: (Optional[str]) -> Any
+            cache_id = hash(filename)
+            if cachepath.exists():
+                try:
+                    with cachepath.open("rb") as fd:
+                        data = pickle.load(fd)
```

Comment:
I think it’s quite dangerous to pickle.load from an arbitrary file. Would it be a possibility to use serialisation to json instead ?

Reply:
@polybassa @guedou This file is local. If you can edit it you can also add code to `.config/scapy/prestart.py` that is malicious.

Considering this I don't see how it's an issue

Reply:
I totally agree on that point. I just wanted to mention it.